catalog/solr/catalog-solr-offline-gazetteer/src/main/java/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolr.java

```diff
@@ -0,0 +1,426 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.solr.offlinegazetteer;
+
+import ddf.catalog.data.types.Core;
+import ddf.catalog.data.types.Location;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrRequest.METHOD;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.SolrPingResponse;
+import org.apache.solr.client.solrj.response.SuggesterResponse;
+import org.apache.solr.client.solrj.util.ClientUtils;
+import org.apache.solr.common.SolrDocument;
+import org.codice.ddf.spatial.geocoding.GeoCodingConstants;
+import org.codice.ddf.spatial.geocoding.GeoEntry;
+import org.codice.ddf.spatial.geocoding.GeoEntryAttributes;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryException;
+import org.codice.ddf.spatial.geocoding.GeoEntryQueryable;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.codice.ddf.spatial.geocoding.context.NearbyLocation;
+import org.codice.solr.client.solrj.SolrClient;
+import org.codice.solr.factory.SolrClientFactory;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.io.WKTReader;
+import org.locationtech.jts.io.WKTWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GazetteerQueryOfflineSolr implements GeoEntryQueryable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GazetteerQueryOfflineSolr.class);
+
+  private static final String CITY_SOLR_QUERY =
+      GeoCodingConstants.CITY_FEATURE_CODES
+          .stream()
+          .map(fc -> String.format("ext.feature-code_txt:%s", fc))
+          .collect(Collectors.joining(" OR ", "(", ")"));
+
+  private static final ThreadLocal<WKTWriter> WKT_WRITER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTWriter::new);
+
+  private static final ThreadLocal<WKTReader> WKT_READER_THREAD_LOCAL =
+      ThreadLocal.withInitial(WKTReader::new);
+
+  protected static final String SUGGEST_Q = "suggest.q";
+  protected static final String SUGGEST_DICT = "suggest.dictionary";
+  protected static final String SUGGEST_PLACE_KEY = "suggestPlace";
```

Comment:
Ironically, `SUGGEST_PLACE_KEY` is the only one thats not a literal key. I called it a key because in my head it was like the value that was the target dictionary. I've revised the name to make more sense.